### PR TITLE
Invoke accessibility actions

### PR DESCRIPTION
**Pull request by adkenyon**
- [x] This is a small change 

**Description:**

I've extended the Actions API to support triggering [Accessibility Actions](https://reactnative.dev/docs/accessibility#accessibility-actions) for both iOS and Android. 

Pardon my Swift/Java 😄

### CODE REVIEW

File: detox/ios/Detox/Invocation/Element.swift [hunk 1]
```diff
@@ -171,6 +171,12 @@ class Element : NSObject {
 		view.dtx_replaceText(text)
 	}
 	
+	func accessibilityAction(_ actionName: String) {
+		let action = view.accessibilityCustomActions!.first(where: { $0.name == actionName })
```
Comment (LeoNatan) [inline detox/ios/Detox/Invocation/Element.swift:175]: Please add checks that the action actually exists before attempting to perform it. If not, fail the test with an explanation why.

File: detox/src/ios/expectTwo.js [hunk 2]
```diff
@@ -154,6 +154,11 @@ class Element {
     return this.withAction('clearText');
   }
 
+  accessibilityAction(actionName) {
```
Comment (LeoNatan) [inline detox/src/ios/expectTwo.js:157]: The API should be named `performAccessibilityAction()`

**Review by LeoNatan — CHANGES_REQUESTED** [review]
Hello,

Thank you for your contribution!
A few small changes required but overall looks good.

Thanks

File: detox/src/ios/expectTwo.js (hunk 2 above)
Comment (adkenyon) [reply detox/src/ios/expectTwo.js:157]: @LeoNatan Just in this place or everywhere I've created a method named `accessibilityAction`?

I had considered  `performAccessibilityAction` originally and passed on it 🤦

Comment (LeoNatan) [reply detox/src/ios/expectTwo.js:157]: I only care about the public API, accessible to Detox users (and docs). The internal stuff doesn't matter, it's descriptive enough.

Comment (LeoNatan) [reply detox/src/ios/expectTwo.js:157]: Thanks

File: docs/APIRef.ActionsOnElement.md [hunk 3]
```diff
@@ -247,15 +247,15 @@ jestExpect(multipleMatchedElements.elements[0].identifier).toBe('FirstElement');
 
 Takes a screenshot containing the matched element. For full details on taking screenshots with Detox, refer to the [screen-shots guide](APIRef.Screenshots.md).
 
-### `accessibilityAction(actionName)`
+### `performAccessibilityAction(actionName)`
 
 Simulates a swipe on the element with the provided options.
```
Comment (LeoNatan) [inline docs/APIRef.ActionsOnElement.md:252]: Probably doesn't simulate a swipe. :)

**Review by LeoNatan — CHANGES_REQUESTED** [review]

File: detox/ios/Detox/Invocation/Element.swift (hunk 1 above)
Comment (LeoNatan) [reply detox/ios/Detox/Invocation/Element.swift]: Fixed this in my commits

**Comment by LeoNatan** [comment]
I pushed some changes directly in the GH UI, and didn't test if it compiles 😂

**Comment by adkenyon** [comment]
Ha, yeah, I was just adding the error checks and saw some unfamiliar merge conflicts. I'll run through and test them now

**Comment by adkenyon** [comment]
Gah, my xcode updated and I'm getting all sorts of other issues. Will try to retest later today or this week

**Comment by LeoNatan** [comment]
Make sure to merge master, so that all the latest iOS 14 and Xcode 12 fixes are in there.

**Comment by adkenyon** [comment]
@LeoNatan, both platforms appear to be working. I had to revert your change to `UIApplication.shared.sendAction`. Let me know if you have anything else in mind there.

I am noticing that the android e2e tests are intermittently failing. Watching the tests run I see action is triggering, so functionally it's working. I suspect the issue is a race between detox asserting that the action is complete and the action actually being completed. Do you have any suggestions to help mitigate this?

**Comment by LeoNatan** [comment]
Let’s wait for @d4vidi ‘s review for Android.

**Comment by LeoNatan** [comment]
@rotemmiz @d4vidi Can you please take a look? Thanks

**Comment by adkenyon** [comment]
@rotemmiz Can you point me in the right direction? I'm unaware of the code generation issue or where to begin. I am happy to dig in and learn it though

**Comment by LeoNatan** [comment]
@rotemmiz Can you please assist here?

**Comment by noomorph** [comment]
I've updated the generated classes. Let's see whether the build passes.

**Comment by noomorph** [comment]
Now there's missing coverage issue. Should be trivial enough hopefully.

**Comment by adkenyon** [comment]
Alright, coverage is fixed

**Comment by adkenyon** [comment]
@LeoNatan I think it's ready

**Comment by noomorph** [comment]
@adkenyon, but the build is red, no? Seems like some issue with the unit tests step.

**Comment by adkenyon** [comment]
Oops, thought I had it squared away locally. I'll check it again in the next day or two.

**Comment by LeoNatan** [comment]
Hello,
Android build seems broken. Any updates here?

**Comment by rotemmiz** [comment]
Totally missed the thread, thanks for waiting though :)
@adkenyon is there still anything you need here? Seems like the Android tests are failing on a few errors, mainly `com.facebook.react.common.JavascriptException: Unsupported top level event type "topAccessibilityAction" dispatched`
Isn't the event name `accessibilityAction` ?

**Comment by adkenyon** [comment]
Apologies for not getting this over the finish line, I've been busy as of late. I had both iOS and Android working weeks ago. 

I'm wondering if `topAccessibilityAction` was removed or changed

**Comment by LeoNatan** [comment]
Ping

**Pull request closed without merging**